Fix wrapping of long room topics (and overlap with apps)

rxl881 opened the pull request:
.mx_RoomHeader_wrapper height needs to be increased to prevent max height of child elements from overflowing allocated space (.mx_RoomHeader_topic.max-height: 42px + .mx_RoomHeader_name.height: 31px == 73px total). Wrapper element height is currently set to 70px.

Setting a column width on the room topic causes overflow to be wrapped to another column (right) and hidden (by existing "overflow: hidden", directive).

lukebarnard1 commented:
So the `column-width` looks great. I'm not convinced we need to change the height of the wrapper, especially as it's already out of line with the RightPanel header. Upon investigation, it seems that
```
.mx_AppsDrawer {
    margin-bottom: 3px;
}
```
is causing the room header to be 3px taller than it should be, which is odd as the app draw was closed. 

I'm not sure why the topic has a max-height tbh.

turt2live commented:
(the 4px thing is mentioned in https://github.com/vector-im/riot-web/issues/4981 fwiw)

rxl881 commented:
Thanks for linking to and reporting the other issue @turt2live. The first part of this (the 3px of text from the top of the wrapped line) should now be fixed:
Before - 
![screen shot 2017-11-08 at 17 50 31](https://user-images.githubusercontent.com/6763606/32564923-8694819e-c4ad-11e7-8b4b-4b725c3a915c.png)

After - 
![screen shot 2017-11-08 at 17 50 44](https://user-images.githubusercontent.com/6763606/32564930-8d191d90-c4ad-11e7-865a-f9bfbf9ad703.png)

I'm a little confused by both of your (slightly differing) reports that there is a problem with the margin or padding of the widgets drawer / tray (would be helpful to have the CSS class / id of the element that you think is a problem in the other issue report @turt2live).

In this case, I am sure that the problem is with .mx_RoomHeader_wrapper. The class .mx_RoomHeader_wrapper has a fixed height of 70px. It contains mx_RoomHeader_name with a fixed height of 31px, and mx_RoomHeader_topic with a max-height of 42px (totalling a hight of 73px, inside a container with a max height of 70px). In the chrome DOM inspector, you can see the child content overflowing onto the content below it:

![screen shot 2017-11-08 at 17 59 43](https://user-images.githubusercontent.com/6763606/32565445-ec01bdf2-c4ae-11e7-9e2b-aa2b4b7ebeea.png)

![screen shot 2017-11-08 at 18 00 14](https://user-images.githubusercontent.com/6763606/32565463-eee2eabe-c4ae-11e7-9bd0-d329b60b9161.png)

An alternative solution to increasing the size of the wrapper is decreasing the max-height of the mx_RoomHeader_topic by 3px.

lukebarnard1 commented:
>An alternative solution to increasing the size of the wrapper is decreasing the max-height of the mx_RoomHeader_topic by 3px.

Increasing the size of the header will only make the mis-alignment with RightPanel worse.

When the app draw is not visible, it shouldn't take up any height. But it is, as described in https://github.com/vector-im/riot-web/issues/4981#issuecomment-338799864.

>(would be helpful to have the CSS class / id of the element that you think is a problem in the other issue report @turt2live).

`.mx_AppsDrawer` is the one you want (as mentioned previously).

rxl881 commented on src/skins/vector/css/matrix-react-sdk/views/rooms/_AppsDrawer.scss:
```diff
@@ -14,9 +14,8 @@ See the License for the specific language governing permissions and
-    margin-bottom: 3px;
-}
+// .mx_AppsDrawer {
+// }
```
I have intentionally left these commented classes for the time being.

rxl881 commented:
Urrghh... I think that we are suffering from some miscommunication here... and are dealing with a couple of (slightly) different issues.

You're absolutely right, there is a margin-bottom, 3px on the .mx_AppsDrawer that shouldn't be there when there are no apps / the apps drawer is not visible. I'll call this issue #2 as it is unrelated to the issue that I was originally trying to solve. I have now fixed this on the most recent commit (by moving the margin to the AppTile elements). This fixes the alignment issues with the left and right columns when no apps are present.

Issue #1 (that this PR is originally about) is still relevant; content overflows .mx_RoomHeader_wrapper due to static height specifications on the contained elements. Regardless of the margin on the apps drawer this causes long topic content to overlap other elements, e.g. the AppsDrawer when apps are present.

@lukebarnard1:
>> An alternative solution to increasing the size of the wrapper is *decreasing* the max-height of the mx_RoomHeader_topic by 3px.

> Increasing the size of the header will only make the mis-alignment with RightPanel worse.

As mentioned in my comment, **decreasing** the max-height will fix the problem without impacting the mis-alignment. I've updated the PR to reflect / implement this.

@lukebarnard1, @ara4n - Can you please check that you're happy, and that this is good to go? Thanks.

lukebarnard1 commented:
>>>An alternative solution to increasing the size of the wrapper is decreasing the max-height of the mx_RoomHeader_topic by 3px.
>>Increasing the size of the header will only make the mis-alignment with RightPanel worse.
>As mentioned in my comment, decreasing the max-height will fix the problem without impacting the mis-alignment. I've updated the PR to reflect / implement this.

Looks like I totally failed to grok what you were saying, @rxl881, sorry about that.

LGTM!

rxl881 commented:
Cool, and no worries at all @lukebarnard1. Thanks! :)
